packages/gatsby/src/utils/babel-loader-helpers.js

```diff
@@ -52,9 +52,14 @@ const prepareOptions = (babel, options = {}, resolve = require.resolve) => {
     ),
   ]
 
+  const develop = stage === `develop`
+  const developPageTemplate = develop && pageTemplate // Matches if import is from async-requires
+  const developOnDemand =
+    develop && process.env.GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND // On demand does not generate async-requires
```

Comment:
That's odd, how does gatsby know what template it needs if it doesn't use async-requires?